File: app/src/main/java/com/example/floating/bakingapp/recipes/RecipeModule.java
package com.example.floating.bakingapp.recipes;

import com.example.floating.bakingapp.data.ApiService;

import dagger.Module;
import dagger.Provides;

/**
 * Created by apple on 14/01/2018.
 */
@Module
public class RecipeModule {
    RecipeContract.View view;

    RecipeModule(RecipeContract.View view){
        this.view = view;
    }

    @Provides
    @RecipeScope
    RecipeContract.View providesRecipeView(){
        return view;
    }

    @Provides
    @RecipeScope
    RecipePresenter providesRecipePresenter(RecipeContract.View view, ApiService apiService){
        return new RecipePresenter(view, apiService);
    }
}

File: app/src/main/java/com/example/floating/bakingapp/recipes/RecipeModule.kt
package com.example.floating.bakingapp.recipes

import com.example.floating.bakingapp.data.ApiService

import dagger.Module
import dagger.Provides

/**
 * Created by apple on 14/01/2018.
 */
@Module
class RecipeModule internal constructor(internal var view: RecipeContract.View) {

    @Provides
    @RecipeScope
    internal fun providesRecipeView(): RecipeContract.View {
        return view
    }

    @Provides
    @RecipeScope
    internal fun providesRecipePresenter(view: RecipeContract.View, apiService: ApiService): RecipePresenter {
        return RecipePresenter(view, apiService)
    }
}
